### PR TITLE
Bluetooth: SDP: fix 32 and 128-bit UUID configuration (IDFGH-10312)

### PR DESCRIPTION
**Pull request by ilutchenko**
## The problem:
`esp_sdp_create_record()` can't create SDP records with 32 and 128-bit UUIDs.

## Summary
Added proper conversion of 4 and 16-byte UUIDs values to binary streams. 
UUIDs are now set with `SDP_AddAttribute()` instead of `SDP_AddServiceClassIdList()`.

Changes were tested with the following code:

```
esp_bluetooth_sdp_record_t record = {};
                
record.hdr.type = ESP_SDP_TYPE_RAW;
record.hdr.uuid.len = 2;
record.hdr.uuid.uuid.uuid16 = 0xfefe;
record.hdr.service_name_length = strlen("16-bit UUID") + 1;
record.hdr.service_name = "16-bit UUID";
record.hdr.rfcomm_channel_number = BT_IAP2_RFCOMM;
record.hdr.l2cap_psm = BT_L2CAP_DYNMIC_PSM;
record.hdr.profile_version = BT_IAP2_PROFILE_VERSION;
esp_sdp_create_record(&record);

record.hdr.type = ESP_SDP_TYPE_RAW;
record.hdr.uuid.len = 4;
record.hdr.uuid.uuid.uuid32 = 0xfefefefe;
record.hdr.service_name_length = strlen("32-bit UUID") + 1;
record.hdr.service_name = "32-bit UUID";
record.hdr.rfcomm_channel_number = BT_IAP2_RFCOMM;
record.hdr.l2cap_psm = BT_L2CAP_DYNMIC_PSM;
record.hdr.profile_version = BT_IAP2_PROFILE_VERSION;
esp_sdp_create_record(&record);

static const uint8_t IAP2_UUID[] = {0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe,
                                    0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe};
               
record.hdr.type = ESP_SDP_TYPE_RAW;
record.hdr.uuid.len = sizeof(IAP2_UUID);
memcpy(record.hdr.uuid.uuid.uuid128, IAP2_UUID, sizeof(IAP2_UUID));
record.hdr.service_name_length = strlen("128-bit UUID") + 1;
record.hdr.service_name = "128-bit UUID";
record.hdr.rfcomm_channel_number = BT_IAP2_RFCOMM;
record.hdr.l2cap_psm = BT_L2CAP_DYNMIC_PSM;
record.hdr.profile_version = BT_IAP2_PROFILE_VERSION;
esp_sdp_create_record(&record);
```

`sdptool scan` outputs will be:

```
Service Name: 16-bit UUID
Service RecHandle: 0x10001
Service Class ID List:
  "" (0xfefe)
Protocol Descriptor List:
  "L2CAP" (0x0100)
  "RFCOMM" (0x0003)
    Channel: 1

Service Name: 32-bit UUID
Service RecHandle: 0x10001
Service Class ID List:
  "" (0xfefefefe)
Protocol Descriptor List:
  "L2CAP" (0x0100)
  "RFCOMM" (0x0003)
    Channel: 1

Service Name: 128-bit UUID
Service RecHandle: 0x10001
Service Class ID List:
  UUID 128: fefefefe-fefe-fefe-fefe-fefefefefefe
Protocol Descriptor List:
  "L2CAP" (0x0100)
  "RFCOMM" (0x0003)
    Channel: 1
```

## Caveats

There are still some strange things like 
`memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid16, sizeof(service));`
I plan to rework them when I will fix [#11528](https://github.com/espressif/esp-idf/issues/11529)

### CODE REVIEW

File: components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c [hunk 1]
```diff
@@ -274,17 +275,25 @@ static int add_raw_sdp(const bluetooth_sdp_record* rec)
...
     if (rec->hdr.bt_uuid.len == 16) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid128[2], sizeof(service));
+        UINT8_TO_BE_STREAM (p_temp, (UUID_DESC_TYPE << 3) | SIZE_SIXTEEN_BYTES);
+        ARRAY_TO_BE_STREAM (p_temp, rec->hdr.bt_uuid.uuid.uuid128, LEN_UUID_128);
     } else if (rec->hdr.bt_uuid.len == 2) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid16, sizeof(service));
+        UINT8_TO_BE_STREAM (p_temp, (UUID_DESC_TYPE << 3) | SIZE_TWO_BYTES);
+        UINT16_TO_BE_STREAM (p_temp, rec->hdr.bt_uuid.uuid.uuid16);
     } else if (rec->hdr.bt_uuid.len == 4) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid16, sizeof(service));
```
Comment (xiongweichao) [inline components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:285]: I want to convert 32 bit or 128 bit uuid to 16 bit uuid.

Comment (ilutchenko) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:285]: Hi!
What is the purpose of converting them to 16-bit UUID?

I dug down and realized that we must patch [bta_sys_add_uuid(UINT16 uuid16)](https://github.com/espressif/esp-idf/blob/a4afa44435ef4488d018399e1de50ad2ee964be8/components/bt/host/bluedroid/bta/sys/bta_sys_conn.c#L528) and other functions in the chain to support 32 and 128 bit UUIDs as well.
I will do that to fix the issue https://github.com/espressif/esp-idf/issues/11529 , but it exceeds this pull request.

File: components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c [hunk 2]
```diff
@@ -274,17 +275,25 @@ static int add_raw_sdp(const bluetooth_sdp_record* rec)
...
+        UINT8_TO_BE_STREAM (p_temp, (UUID_DESC_TYPE << 3) | SIZE_FOUR_BYTES);
+        UINT32_TO_BE_STREAM (p_temp, rec->hdr.bt_uuid.uuid.uuid32);
     } else {
         SDP_DeleteRecord(sdp_handle);
         sdp_handle = 0;
         return sdp_handle;
     }
     /* add service class */
-    status &= SDP_AddServiceClassIdList(sdp_handle, 1, &service);
+    status &= SDP_AddAttribute (sdp_handle, ATTR_ID_SERVICE_CLASS_ID_LIST,
```
Comment (xiongweichao) [inline components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:294]: Please delete the trailing-whitespace.

File: components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c (hunk 1 above)
Comment (xiongweichao) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:285]: Maybe we don't need to modify `bta_sys_add_uuid`. Just need to convert them to 16 bit uuid.

Comment (ilutchenko) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:285]: That is not the topic of this PR, but I want to explain myself:

Bluedroid in some places has the ability to use 128-bit UUID.
For example, in the [bta_dm_set_eir()](https://github.com/espressif/esp-idf/blob/3a45d4e949a174e8829a2e4c86c421b030ceac5a/components/bt/host/bluedroid/bta/dm/bta_dm_act.c#L4212) code checks for the 128-bit UUIDs.
But the problem is that these 128 and 32-bit UUIDs are nowhere saved in the [bta_dm_cb.custom_uuid[] ](https://github.com/espressif/esp-idf/blob/3a45d4e949a174e8829a2e4c86c421b030ceac5a/components/bt/host/bluedroid/bta/dm/bta_dm_act.c#L4222)array. Only 16-bit values are stored.
That's why I want to modify `bta_sys_add_uuid` one day.

File: components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c (hunk 2 above)
Comment (xiongweichao) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c]: I'm very sorry, maybe I expressed it wrong. It is necessary to delete the space behind `,`.

Comment (ilutchenko) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c]: Oh, got it.
Fixed!

File: components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c (hunk 1 above)
Comment (xiongweichao) [reply components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c:285]: you are right. In order to support 128 and 32-bit UUIDs need to modify `bta_sys_add_uuid`.

**Pull request merged**